batcher/aligned-batcher/src/retry/batcher_retryables.rs

```diff
@@ -60,29 +87,140 @@ pub async fn user_balance_is_unlocked_retryable(
+        }
+        _ => {
+            call_fallback = payment_service_fallback
+                .create_new_task(
+                    batch_merkle_root,
+                    batch_data_pointer,
+                    proofs_submitters,
+                    fee_params.fee_for_aggregator,
+                    fee_params.fee_per_proof,
+                    fee_params.respond_to_task_fee_limit,
+                )
+                .gas_price(fee_params.gas_price);
+            match call_fallback.send().await {
+                Ok(pending_tx) => pending_tx,
+                Err(ContractError::Revert(err)) => {
+                    warn!("Transaction reverted {:?}", err);
+                    return Err(RetryError::Permanent(BatcherError::TransactionSendError));
+                }
+                _ => return Err(RetryError::Transient(BatcherError::TransactionSendError)),
+            }
+        }
+    };
+
+    // timeout to prevent a deadlock while waiting for the transaction to be included in a block.
+    timeout(Duration::from_millis(transaction_wait_timeout), pending_tx)
+        .await
+        .map_err(|e| {
+            warn!("Error while waiting for batch inclusion: {e}");
+            RetryError::Permanent(BatcherError::ReceiptNotFoundError)
+        })?
+        .map_err(|e| {
+            warn!("Error while waiting for batch inclusion: {e}");
+            RetryError::Transient(BatcherError::TransactionSendError)
+        })?
+        .ok_or(RetryError::Permanent(BatcherError::ReceiptNotFoundError))
+}
+
+pub async fn cancel_create_new_task_retryable(
+    batcher_signer: &SignerMiddlewareT,
+    batcher_signer_fallback: &SignerMiddlewareT,
+    bumped_gas_price: U256,
+    transaction_wait_timeout: u64,
+) -> Result<TransactionReceipt, RetryError<ProviderError>> {
+    let batcher_addr = batcher_signer.address();
+
+    let current_nonce = get_current_nonce(
+        batcher_signer.provider(),
+        batcher_signer_fallback.provider(),
+        batcher_addr,
+    )
+    .await
+    .map_err(RetryError::Transient)?;
+
+    let tx = TransactionRequest::new()
+        .to(batcher_addr)
+        .value(U256::zero())
+        .nonce(current_nonce)
+        .gas_price(bumped_gas_price);
+
+    let pending_tx = match batcher_signer.send_transaction(tx.clone(), None).await {
+        Ok(pending_tx) => pending_tx,
+        Err(_) => batcher_signer_fallback
+            .send_transaction(tx.clone(), None)
+            .await
+            .map_err(|e| RetryError::Transient(ProviderError::CustomError(e.to_string())))?,
+    };
+
+    // timeout to prevent a deadlock while waiting for the transaction to be included in a block.
+    timeout(Duration::from_millis(transaction_wait_timeout), pending_tx)
+        .await
+        .map_err(|e| {
+            warn!("Timeout while waiting for transaction inclusion: {e}");
+            RetryError::Transient(ProviderError::CustomError("Receipt not found".to_string()))
```

Comment:
Wouldn't the error be something timeout related here?

Reply:
You want me to improve the description of the error here?

Reply:
Yes.

Reply:
Done!